staging/src/k8s.io/kubectl/pkg/cmd/debug/profiles.go

```diff
@@ -57,3 +70,162 @@ func profileLegacy(pod *corev1.Pod, containerName string, target runtime.Object)
 		return fmt.Errorf("the %s profile doesn't support objects of type %T", ProfileLegacy, target)
 	}
 }
+
+func (p *generalProfile) Apply(pod *corev1.Pod, containerName string, target runtime.Object) error {
+	switch target.(type) {
+	case *corev1.Pod:
```

Comment:
For the case of pod we'll want to detect whether `pod` and `target` are the same and have different behavior. I don't think we plan on stripping labels for debug by ephemeral container.

Reply:
hmm I would like to double check, the proposal https://github.com/kubernetes/enhancements/tree/master/keps/sig-cli/1441-kubectl-debug#profile-general says that:
> Probes and labels are stripped from Pod copies to ensure the copy isn't killed and doesn't receive traffic during debugging.

Do we need to strip labels?

Reply:
When debugging a pod, there are two outcomes:

1. Creating a copy of the pod. This should strip labels & probes.
2. Adding an ephemeral container to an existing pod. This should not change labels & probes.

I should have been more clear in the KEP. I meant for the section you pasted to apply to "Pod copies" (i.e. the first case above)